step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/execution/LocalAutomationPackageRepository.java

```diff
@@ -52,14 +57,42 @@ public ArtefactInfo getArtefactInfo(Map<String, String> repositoryParameters) th
         String apName = repositoryParameters.get(AP_NAME);
 
         ArtefactInfo info = new ArtefactInfo();
-        info.setType("automationPackage");
-        info.setName(apName);
+        boolean isWrapInTestSet = isWrapPlansIntoTestSet(repositoryParameters);
+        //Introduced for 28.1, previous execution do not have this field in repo parameters
+        String rootType = repositoryParameters.get(ArtifactRepositoryConstants.PARAM_ROOT_TYPE);
+        if (rootType != null) {
+            info.setType(rootType);
+        } else {
+            info.setType((isWrapInTestSet) ? TestSet.class.getSimpleName() : TestCase.class.getSimpleName());
+        }
+        info.setName((isWrapInTestSet) ? apName : repositoryParameters.getOrDefault(ArtifactRepositoryConstants.PARAM_INCLUDE_PLANS, apName));
         return info;
     }
 
     @Override
     public TestSetStatusOverview getTestSetStatusOverview(Map<String, String> repositoryParameters, ObjectPredicate objectPredicate) throws Exception {
-        return new TestSetStatusOverview();
+        TestSetStatusOverview testSetStatusOverview = new TestSetStatusOverview();
+        if (isWrapPlansIntoTestSet(repositoryParameters)) {
+            PackageExecutionContext ctx = null;
+            try {
+                ctx = getOrRestorePackageExecutionContext(repositoryParameters, null, objectPredicate);
+                //If wrap we return all plans of the AP
+                List<TestRunStatus> runs = getFilteredPackagePlans(ctx.getAutomationPackage(), repositoryParameters, ctx.getAutomationPackageManager())
+                        .map(plan -> new TestRunStatus(getPlanName(plan), getPlanName(plan), ReportNodeStatus.NORUN)).collect(Collectors.toList());
```

Comment:
If the data are available, it would be nice to also set the status of the latest execution